logexporter/cluster/log-dump.sh

```diff
@@ -853,6 +852,47 @@ function main() {
     echo "Dumping logs from nodes locally to '${report_dir}'"
     dump_nodes
   fi
+}
+
+# If 'LOGDUMP_TO_GCS_ONLY' is set, it will dump all logs (including master and nodes fetched by SSH) to ${gcs_artifacts_dir}.
+# This way, ${gcs_artifacts_dir} can be set to a different value than the target where artifacts/ will be uploaded and all logs
+# will be uploaded to the same dir regardless of the way the logs are dumped.
+function main() {
+  setup
+  kube::util::ensure-temp-dir
+  # Copy master logs to artifacts dir locally (through SSH).
+  if [[ -n "${LOGDUMP_TO_GCS_ONLY:-}" ]]; then
+    report_dir="${KUBE_TEMP}/logs"
+    mkdir -p "${report_dir}"
+    echo "${gcs_artifacts_dir}" > "${local_report_dir}/master-and-node-logs.txt"
+    echo "Dumping logs temporarily to '${report_dir}'. Will upload to '${gcs_artifacts_dir}' later."
+  else
+    report_dir="${local_report_dir}"
+    echo "Dumping logs locally to '${local_report_dir}'"
+  fi
+
+  dump_logs
+
+  if [[ -n "${LOGDUMP_TO_GCS_ONLY:-}" ]]; then
+    echo "Uploading '${report_dir}' to '${gcs_artifacts_dir}'"
+
+    # Normally, result of gsutil uploading data will be different when target directory exists
+    # and when it's missing. `gsutil cp -r /tmp/dir/logs gs://target-bucket/X/1234/`:
+    # 1. will create `gs://target-bucket/X/1234/logs` if '1234' existed.
```

Comment:
just for my own education: it will just create the directory and not upload it?

I think I don't fully understand the difference between 1 and 2 here.